scripts/ai/verify-ai-compat.sh

```
@@ -1469,6 +1472,169 @@ _check_pinning_policy_text "$REPO_ROOT/modules/shared/programs/claude/files/CLAU
+    [ -n "$_candidate" ] || continue
+    # 자기 자신 (lib 정의 파일) 제외.
+    [ "$_candidate" = "$_lib_path" ] && continue
+    # lib 자체 디렉토리 내 다른 lib 파일은 제외 (예: hook-runtime 검증 시 pinning-patterns 본문 매치).
+    case "$_candidate" in
+      */modules/shared/programs/claude/files/lib/*) continue ;;
+      */modules/shared/programs/codex/files/lib/*) continue ;;
+    esac
+    # 변수 할당 + source 호출 패턴 둘 다 있어야 진짜 후보.
+    if grep -qE "^[[:space:]]*[A-Z_]+_LIB=.*${_lib_basename_re}" "$_candidate" \
+      && grep -qE "(\.[[:space:]]+|source[[:space:]]+)\"\\\$[A-Z_]+_LIB\"" "$_candidate"; then
+      _candidate_abs_rel="${_candidate#$REPO_ROOT/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)